sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -482,7 +482,7 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
 
     case CreateViewStatement(
       tbl, userSpecifiedColumns, comment, properties,
-      originalText, child, allowExisting, replace, viewType) if child.resolved =>
```

Comment:
`if child.resolved` is no longer needed because the `child` becomes the child of `CreateViewCommand`.

Reply:
where do we check `CreateViewCommand.children` is resolved?

Reply:
`CreateViewCommand.children` should be resolved by the Analyzer automatically. Do we need an extra check?

Reply:
I think so, because `ResolveSessionCatalog` is also an analyzer rule and there is no guarantee that the plans we hit here are resolved.

Reply:
We needed this check because `CreateViewCommand.plan` was not a real child before. Without this check, you could potentially convert `CreateViewStatement` to `CreateViewCommand` even when `CreateViewStatement.child` was not resolved yet, and it will remain unresolved since `CreateViewCommand` didn't have `children`.

Now that `CreateViewCommand.plan` is a real child, I don't think we need the check any longer since it will be resolved by the analyzer again, no?

Reply:
Ah sorry I misread the code. Seems we can do the same in `AlterViewAsCommand` and remove `if query.resolved` in the above case?

Reply:
good catch!